### PR TITLE
feat(e2e): manage ica from gov module

### LINKED ISSUE #1746: Create Gov mediated ICA E2E Test
## Problem Definition

Once `icamauth` is in main we will be able to test the entire user flow of creating and controlling an interchain account using the new gov module. The current gov E2E and the current icamauth E2E do not contain this flow.

We should make a new E2E test that focuses on the goveranance use case of interchain accounts.

## Proposal

Outline the exact flow based on @hxrts input.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

### PR DESCRIPTION
**Pull request by Pantani**
closes #1746

### CODE REVIEW

File: tests/e2e/e2e_gov_test.go [hunk 1]
```diff
@@ -120,36 +121,110 @@ Test Benchmarks:
...
+3. Check if ICA account exist
+*/
+func (s *IntegrationTestSuite) GovCreateICA() {
+	s.Run("test create ica from gov module", func() {
+		var (
+			portID          = "1317/tcp"
+			validatorChainA = s.chainA.validators[0]
+			validatorChainB = s.chainB.validators[0]
+			resourceChainA  = s.valResources[s.chainA.id][0]
+			resourceChainB  = s.valResources[s.chainA.id][0]
```
Comment (okwme) [inline tests/e2e/e2e_gov_test.go]: why `chainA` twice?

File: tests/e2e/e2e_setup_test.go [hunk 2]
```diff
@@ -66,12 +79,18 @@ const (
...
 	govProposalBlockBuffer       = 35
 	relayerAccountIndex          = 0
-	icaOwnerAccountIndex         = 1
 	numberOfEvidences            = 10
 	slashingShares         int64 = 10000
 )
 
+const (
+	// genesis accounts enum
+	icaOwnerAccountIndex = iota + 1
```
Comment (okwme) [inline tests/e2e/e2e_setup_test.go:89]: ooo i didn't know about this : )

File: tests/e2e/e2e_gov_test.go (hunk 1 above)
Comment (okwme) [reply tests/e2e/e2e_gov_test.go]:
```suggestion
			resourceChainB  = s.valResources[s.chainB.id][0]
```

File: tests/e2e/e2e_ibc_test.go [hunk 3]
```diff
@@ -140,7 +140,7 @@ func (s *IntegrationTestSuite) sendIBC(c *chain, valIdx int, sender, recipient,
 		"ibc-transfer",
 		"transfer",
 		"transfer",
-		"channel-0",
+		icaChannelID,
```
Comment (yaruwangway) [inline tests/e2e/e2e_ibc_test.go]: this is not related to ica, we can rename `icaChannelID` -> `ibcTransferChannelID`

**Review by yaruwangway — APPROVED** [review]

**Review by okwme — APPROVED** [review]
